src/base.rs

```diff
@@ -0,0 +1,290 @@
+///         guarded_transmute_many::<u16, SingleManyGuard>(&[0x00, 0x01, 0x00, 0x02])?,
+/// # */
+/// #   assert_eq!(guarded_transmute_many::<u16, SingleManyGuard>(&[0x00, 0x01, 0x00, 0x02].le_to_native::<u16>()).unwrap(),
+///         &[0x0100, 0x0200]
+///     );
+/// }
+/// # }
+/// ```
+pub unsafe fn guarded_transmute_many<T, G>(bytes: &[u8]) -> Result<&[T], Error>
+where
+    G: Guard,
+{
+    G::check::<T>(bytes)?;
+    Ok(slice::from_raw_parts(bytes.as_ptr() as *const T, bytes.len() / size_of::<T>()))
+}
+
+/// View a byte slice as a slice of an arbitrary type.
+///
+/// The resulting slice will have as many instances of a type as will fit,
+/// rounded down. The permissive guard is a no-op, which makes it possible for
+/// this function to return a slice directly. It is therefore equivalent to
+/// `guarded_transmute_many::<_, PermissiveGuard>(bytes).unwrap()`.
+///
+/// # Safety
+///
+/// - This function does not perform memory alignment checks. The beginning of
+/// the slice data must be properly aligned for accessing vlues of type `T`.
+/// - The byte data needs to correspond to a valid contiguous sequence of `T`
+/// values. Types `T` with a `Drop` implementation are unlikely to be safe
+/// in this regard.
+///
+/// Failure to fulfill any of the requirements above is dangerous and may eat
+/// your laundry.
+///
+/// # Examples
+///
+/// ```
+/// # use safe_transmute::base::guarded_transmute_many_permissive;
+/// # include!("../tests/test_util/le_to_native.rs");
+/// # fn main() {
+/// // Little-endian
+/// unsafe {
+/// # /*
+///     assert_eq!(
+///         guarded_transmute_many_permissive::<u16>(&[0x00, 0x01, 0x00, 0x02]),
+/// # */
+/// #   assert_eq!(guarded_transmute_many_permissive::<u16>(&[0x00, 0x01, 0x00, 0x02].le_to_native::<u16>()),
+///         &[0x0100, 0x0200]
+///     );
+/// }
+/// # }
+/// ```
+pub unsafe fn guarded_transmute_many_permissive<T>(bytes: &[u8]) -> &[T] {
+    guarded_transmute_many::<_, crate::guard::PermissiveGuard>(bytes).expect("permissive guard should never fail")
+}
+
+/// Transform a byte vector into a vector of an arbitrary type.
+///
+/// The vector's allocated byte buffer (if already allocated) will be reused.
+/// The required length of the vector depends on the chosen boundary guard.
+/// Please see the [Guard API](../guard/index.html).
+///
+/// # Safety
+///
+/// - This function does not perform memory alignment checks. The beginning of
+/// the slice data must be properly aligned for accessing vlues of type `T`.
+/// - The byte data needs to correspond to a valid contiguous sequence of `T`
+/// values. Types `T` with a `Drop` implementation are unlikely to be safe
+/// in this regard.
+///
+/// Failure to fulfill any of the requirements above is dangerous and may eat
+/// your laundry.
+///
+/// # Examples
+///
+/// ```
+/// use safe_transmute::guard::PermissiveGuard;
+/// # use safe_transmute::base::guarded_transmute_vec;
+/// # include!("../tests/test_util/le_to_native.rs");
+///
```

Comment:
Stray newline.

Occurs on:
  * L210
  * L265

Reply:
Is this still applicable? Can't seem to find those.

Reply:
'S a'igh', I'll get'em when doing Pedantic Pass later